src/Definition/Reflection/Parameter.php

```diff
@@ -80,11 +85,24 @@ public function isRequired(): bool
      * {@inheritDoc}
      *
      * @see ParameterInterface::isScalar()
+     *
+     * @throws UnexpectedValueException
      */
     public function isBuiltin(): bool
     {
         if ($this->reflection->hasType()) {
             $type = $this->reflection->getType();
+
+            if ($type !== null && ! $type instanceof ReflectionNamedType) {
+                throw new UnexpectedValueException(
```

Comment:
This isn't really "unexpected" - it is very much expected.

This is more of a "known limitation", for which we should (IMO) design a new exception class.

In addition to that:

 * add a named constructor to that exception
 * we need to explain (in the wording) why/where something isn't supported yet

Reply:
What do you think of the newly added test class and its named constructor?

Reply:
excellent!

Reply:
Wait, I'm sorry for being early but would `UnsupportedReflectionType::class` be better?

Reply:
I updated the name to `UnsupportedReflectionTypeException::class` sorry for that. Since both `ReflectionUnionType::class` and `ReflectionIntersectionType::class` extend `ReflectionType::class`.